thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/datasource/loader/DefaultTimeSeriesLoader.java

```diff
@@ -36,16 +37,18 @@
 
   private final MetricConfigManager metricDAO;
   private final DatasetConfigManager datasetDAO;
-  private final QueryCache cache;
+  private final QueryCache queryCache;
+  private final DefaultTimeSeriesCache timeSeriesCache;
```

Comment:
My suggestion here is that we should put `timeSeriesCache` in `DefaultDataProvider`, as this cache is in the same level as the memory cache we had, from the perspective of the architecture. It is much easier to understand if all the logic of loading from cache or data source is in one place.